### PR TITLE
 #53-css-variables-to-scss 

### LINKED ISSUE #53: CSS Variables to SCSS

### PR DESCRIPTION
**Pull request by alisherj0n**

### CODE REVIEW

File: src/_base.scss [hunk 1]
```diff
@@ -1,6 +1,8 @@
+@use "./colors" as *;
```
Comment (rasulovdev) [inline src/_base.scss]: What does `as *` do? Currently you're just making it work, without using the full advantage. Read the docs carefully about what it is used for and strictly follow the docs.

File: src/_colors.scss [hunk 2]
```diff
@@ -1,27 +1,25 @@
...
+$color-danger-light: rgba(253, 121, 114, 0.1);
+$color-danger-dark: #ea2b21;
+$color-warning: #ff965d;
+$color-warning-hover: #ffa778;
+$color-warning-light: rgba(229, 110, 25, 0.1);
+$color-warning-dark: #e56e19;
+$color-success: #49c96d;
+$color-success-hover: #79cc91;
+$color-success-light: rgba(73, 201, 109, 0.1);
+$color-success-dark: #27b950;
```
Comment (rasulovdev) [inline src/_colors.scss]: In combination with **_correct_** use of `@use "..." as ...` we no longer need to prefix our variables with their type – `color-` in this case.

File: src/_base.scss (hunk 1 above)
Comment (rasulovdev) [reply src/_base.scss]: Just read `@use "./colors" as *`. It is a plain english. You can tell what `as` keyword does right away.

**Review by rasulovdev — CHANGES_REQUESTED** [review]

File: src/_typography.scss [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+$font-family: "Poppins", sans-serif;
 
 h1,
 h2,
 h3,
 h4,
 h5,
 h6 {
-  color: var(--color-heading);
+  color: map-get(colors.$color, "heading");
```
Comment (rasulovdev) [inline src/_typography.scss]: When you do maps, there's no need in namespaces. Don't be dumb to repeat the same mistake for 1234598023495 times. What else could you pull from `colors` if not `color`? It should be `map-get(colors, "heading")`. Same for all maps

File: src/_base.scss [hunk 4]
```diff
@@ -1,17 +1,21 @@
+@use "./colors";
+@use "./fx";
+@use "./typography";
+
 body {
-  font-family: var(--font-family);
-  color: var(--color-body);
+  font-family: typography.$font-family;
+  color: map-get(colors.$color, "body");
```
Comment (rasulovdev) [inline src/_base.scss]: When you do maps, there's no need in namespaces. Don't be dumb to repeat the same mistake for 1234598023495 times. What else could you pull from colors if not color? It should be map-get(colors, "body"). Same for all maps.

Comment (rasulovdev) [reply src/_base.scss]: A map is a virtual file itself. You can combine all maps in a single file and call it `variables.scss` without the risk of leaking anything.

File: src/_colors.scss [hunk 5]
```diff
@@ -1,27 +1,27 @@
...
-  --color-warning: #ff965d;
-  --color-warning-hover: #ffa778;
-  --color-warning-light: rgba(229, 110, 25, 0.1);
-  --color-warning-dark: #e56e19;
-  --color-success: #49c96d;
-  --color-success-hover: #79cc91;
-  --color-success-light: rgba(73, 201, 109, 0.1);
-  --color-success-dark: #27b950;
-}
+$color: (
```
Comment (rasulovdev) [inline src/_colors.scss]: There isn't a single color

File: src/_fx.scss [hunk 6]
```diff
@@ -1,7 +1,12 @@
...
+$radius-sm: 10px;
+$radius-xs: 6px;
+$transition-duration: 0.2s;
+
+$radius: (
+  "xs": $radius-xs,
+  "sm": $radius-sm,
+  "radius": $radius,
+  "xl": $radius-xl,
+);
```
Comment (rasulovdev) [inline src/_fx.scss]: This is bullshit. You just doubled the code size

File: src/_form-controls.scss [hunk 7]
```diff
@@ -1,7 +1,5 @@
...
-  --form-control-padding-y: 0.4628rem;
-  --form-control-height: 2.625rem;
-  --form-control-height-lg: 3rem;
-  --form-control-height-sm: 2.25rem;
-}
+$padding-x: 0.8rem;
+$padding-y: 0.4628rem;
+$height: 2.625rem;
+$height-lg: 3rem;
+$height-sm: 2.25rem;
```
Comment (rasulovdev) [inline src/_form-controls.scss]: any thoughts?

Comment (rasulovdev) [reply src/_form-controls.scss]: for lines 3-5

File: src/_layout.scss [hunk 8]
```diff
@@ -1,6 +1,6 @@
-:root {
-  --container-padding: 1.2rem;
-}
+@use "./breakpoints";
+
+$container-padding: 1.2rem;
```
Comment (rasulovdev) [inline src/_layout.scss]: <s>Make sure you're not using this var anywhere outside. If you do, then extract it into a higher level file, same as we discussed about form-controls vars being used in buttons</s>

YOU ARE USING IT OUTSIDE.

File: src/components/_app-header.scss [hunk 9]
```diff
@@ -1,6 +1,8 @@
-:root {
-  --app-header-height: 66px;
-}
+@use "../colors";
+@use "../breakpoints";
+@use "../layout";
```
Comment (rasulovdev) [inline src/components/_app-header.scss]: Bullshit.

**Review by rasulovdev — CHANGES_REQUESTED** [review]

File: src/_variables.scss [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  "sm": 768px,
+  "lg": 992px,
+  "xl": 1200px,
+);
+
+$form-controls: (
+  "height": 2.625rem,
+  "height-lg": 3rem,
+  "height-sm": 2.25rem,
+);
```
Comment (rasulovdev) [inline src/_variables.scss]: bullshit

File: src/_variables.scss [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  "sm": 10px,
+  "xs": 6px,
+);
+
+$breakpoints: (
+  "xs": 576px,
+  "sm": 768px,
+  "lg": 992px,
+  "xl": 1200px,
+);
```
Comment (rasulovdev) [inline src/_variables.scss]: Why sizes' order is different in different maps?

File: src/_variables.scss [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+$form-controls: (
+  "height": 2.625rem,
+  "height-lg": 3rem,
+  "height-sm": 2.25rem,
+);
+
+$container-padding: 1.2rem;
+$transition-duration: 0.2s;
```
Comment (rasulovdev) [inline src/_variables.scss]: If you decided to do this, then why other **_shared_** vars are in different files?

**Review by rasulovdev — CHANGES_REQUESTED** [review]

File: src/_variables.scss (hunk 12 above)
Comment (rasulovdev) [reply src/_variables.scss]: This is not resolved

**Review by rasulovdev — CHANGES_REQUESTED** [review]
🤦‍♂️

File: src/_variables.scss [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+  "xs": 576px,
+  "sm": 768px,
+  "lg": 992px,
+  "xl": 1200px,
+);
+
+$form-controls: (
+  "height-xs": 2.25rem,
+  "height-sm": 2.625rem,
+  "height-lg": 3rem,
```
Comment (rasulovdev) [inline src/_variables.scss]: b

Comment (rasulovdev) [reply src/_variables.scss]: o

Comment (rasulovdev) [reply src/_variables.scss]: o

Comment (rasulovdev) [reply src/_variables.scss]: l

Comment (rasulovdev) [reply src/_variables.scss]: s

Comment (rasulovdev) [reply src/_variables.scss]: h

Comment (rasulovdev) [reply src/_variables.scss]: e

Comment (rasulovdev) [reply src/_variables.scss]: e

Comment (rasulovdev) [reply src/_variables.scss]: t

**Review by rasulovdev — CHANGES_REQUESTED** [review]

File: src/_variables.scss [hunk 14]
```diff
@@ -0,0 +1,82 @@
+$colors: (
```
Comment (rasulovdev) [inline src/_variables.scss:1]: Find and follow my previous comments about **_what_** variables should be extracted into this file.

PS. **This comment is NOT related to colors.** You just ignored my prev comments, so I won't point the exact vars.

PPS. When I say "... _**what**_ variables...", it means "there're variable(s), that shouldn't be there"

File: src/_variables.scss [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+  "xs": 576px,
+  "sm": 768px,
+  "lg": 992px,
+  "xl": 1200px,
+);
+
+$form-controls: (
+  "height-xs": 2.25rem,
+  "height-sm": 2.625rem,
+  "height-lg": 3rem,
```
Comment (rasulovdev) [inline src/_variables.scss]: Bullshit. Season 1, Episode 4.

**Review by rasulovdev — CHANGES_REQUESTED** [review]

File: src/_variables.scss [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+  "lg": 1.7rem,
+);
+
+$container-padding: 1.2rem;
+$container-padding-xl: 1.8rem;
+$transition-duration: 0.2s;
+$font-family: "Poppins", sans-serif;
+$form-controls-padding-x: 0.8rem;
+$form-controls-padding-y: 0.4628rem;
+$form-controls-padding-y-xl: 2rem;
```
Comment (rasulovdev) [inline src/_variables.scss]: I see e new episode of bullshit is coming...

**Review by rasulovdev — CHANGES_REQUESTED** [review]

File: src/_variables.scss [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+$form-controls-padding-y: (
+  "sm": 0.4628rem,
+  "lg": 2rem,
+);
+
+$container-padding: 1.2rem;
+$container-padding-xl: 1.8rem;
+$transition-duration: 0.2s;
+$font-family: "Poppins", sans-serif;
+$form-controls-padding-x: 0.8rem;
```
Comment (rasulovdev) [inline src/_variables.scss]: Why have a map and a plain var for the same property?

File: src/_variables.scss [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+  "lg": 3rem,
+);
+
+$form-controls-padding-y: (
+  "sm": 0.4628rem,
+  "lg": 2rem,
+);
+
+$container-padding: 1.2rem;
+$container-padding-xl: 1.8rem;
```
Comment (rasulovdev) [inline src/_variables.scss]: ?

Comment (rasulovdev) [reply src/_variables.scss]: 54-55

File: src/_variables.scss [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+$transition-duration: 0.2s;
+$font-family: "Poppins", sans-serif;
```
Comment (rasulovdev) [inline src/_variables.scss:64]: Why is this here?

**Review by rasulovdev — CHANGES_REQUESTED** [review]

File: src/_variables.scss (hunk 19 above)
Comment (alisherj0n) [reply src/_variables.scss:64]: This var is used in several places like base, normalize and form-controls

**Pull request merged**